### ISSUE TITLE
Critical performance bug

### ISSUE THREAD

**Issue by ruscon** [timber/timber#1488]
Timber 1.3.4
with woocommerce 6k posts in db
query monitor plugin

testing page with posts and pagination

**test №1 (hprof enabled)**

web/app/plugins/timber-library/lib/PostGetter.php
1) with default get_post_class function

page load ~19s

2) with custom get_post_class function

```php
	/**
	 * @param string $post_type
	 * @param string|array $post_class
	 *
	 * @return string
	 */
	public static function get_post_class( $post_type, $post_class = '\Timber\Post' ) {
		$post_class = apply_filters('Timber\PostClassMap', $post_class);
		$post_class_use = '\Timber\Post';

		if ( is_array($post_class) ) {
			if ( isset($post_class[$post_type]) ) {
				$post_class_use = $post_class[$post_type];
			} else {
				Helper::error_log($post_type.' not found in '.print_r($post_class, true));
			}
		} elseif ( is_string($post_class) ) {
			$post_class_use = $post_class;
		} else {
			Helper::error_log('Unexpeted value for PostClass: '.print_r($post_class, true));
		}

//		$test_post = false;
//		if ( class_exists($post_class_use) ) {
//			$test_post = new $post_class_use();
//		}
//		if ( !$test_post || !(is_subclass_of($test_post, '\Timber\Post') || is_a($test_post, '\Timber\Post')) ) {
//			Helper::error_log('Class ' . $post_class_use . ' either does not exist or implement \Timber\Post');
//		}

		return $post_class_use;
	}
```

page load ~10.5s!

**test №2 (hprof disabled)**

1) with default get_post_class function

page load ~12.5s

2) with custom get_post_class function

page load ~6s!

**Conclusion:** 2x speed

**Comment by ruscon** [comment]
yep

**Comment by ruscon** [comment]
And another vary slow function is in 
TimberPost::post_class()
http://take.ms/Hk18E

Is it need called every time in TimberPost::init() ?
Boots 2x if it not calling :)

**Comment by ruscon** [comment]
**Conclusion:** if we refactoring this 2 function = 4x boost!

**Comment by addedlovely** [comment]
Likewise, saw a drop of about 6 seconds on a slow loading page (composite product builder with currency switcher).

**Comment by jarednova** [comment]
Thanks again to @ruscon to the note and fix. This is resolved in #1522
